### ISSUE TITLE
SBAS: Dimension don't match!

### ISSUE THREAD

**Issue by Mieszko-Fydrych** [gmtsar/gmtsar#18]
Hello,

When running sbas, I get the error "dimension don't match! ../merge/2018220_2018268/corr.grd"
When I run gmt grdinfo on that file, I notice that the xdim and ydim are significantly smaller on the unwrap.grd file than on phasefilt.grd or corr.grd. corr.grd and phasefilt.grd have dimensions of 8618x1839 while unwrap.grd has dimensions of 1077x920.

Did something go wrong with my unwrapping step? 

My unwrap_intf (converted to txt file) is attached. 
[unwrap_intf.txt](https://github.com/gmtsar/gmtsar/files/4315318/unwrap_intf.txt)

Thank you,
Mieszko

**Comment by Xiaohua-Eric-Xu** [comment]
Hi Mieszko,
1077x920 is the grid dimension while 8618x1839 is the range/azimuth dimension. You need to multiply the decimation factor. Since now gmtsar does integer decimation only so the output will be probably 8616x1840. Double check your grids with gmt grdinfo.
Xiaohua(Eric)

**Comment by Mieszko-Fydrych** [comment]
Hi Xiaohua,

I ran it once more, using 8616x1840 as xdim and ydim, but get the same error, now on my corr.grd file. I ran the script using the following command: 

> sbas intf.tab scene.tab 10 5 8616 1840 -range 855042 -incidence 40 -wavelength 0.0554658 -smooth -rms -dem

I got the following output: 

> compute DEM error
> 
> setting smoothing to   0.000 
> setting radar wavelength to   0.055 m 
> setting radar incidence angle to  40.000 degree 
> setting range to 855042.000 m 
> Memory Allocation Successful...
> read table file ...
> number of SAR scenes is 5 
> number of interferograms is 10 
> read phase and correlation grids ...
>  dimension don't match! ../merge/2018220_2018268/corr.grd 


I included screenshots of the gmt grdinfo output for my unwrap.grd and corr.grd files. 

![corr](https://user-images.githubusercontent.com/61254441/76990527-7587f080-691e-11ea-909d-12a869e47536.PNG)
![unwrap](https://user-images.githubusercontent.com/61254441/76990528-7587f080-691e-11ea-8aeb-34b9957b2207.PNG)

My files within /SBAS/ are also attached.

[intf.txt](https://github.com/gmtsar/gmtsar/files/4350479/intf.txt)
[scene.txt](https://github.com/gmtsar/gmtsar/files/4350480/scene.txt)

So, am I messing up the merging/unwrapping step somewhere? Or am I just misunderstanding the sbas inputs? 

Thanks for the help,
Mieszko

**Comment by Xiaohua-Eric-Xu** [comment]
Hi Mieszko,
It's a bit strange that your unwrap.grd and corr.grd has different size. What did you do to produce these? When you run snaphu.csh, did you add the region_cut parameter?  
Xiaohua(Eric)

**Comment by Mieszko-Fydrych** [comment]
Hi Xiaohua,

I produced these results by creating corr_stack.grd using
> ls 201*/corr.grd > corr.grd_list
> stack.csh corr.grd_list 1 corr_stack.grd std.grd

Then created an ocean mask using

> gmt grdmath corr_stack.grd 0.075 GE 0 NAN = mask_def.grd

Set up my unwrap_intf.csh as:

> #!/bin/csh -f
# intflist contains a list of all date1_date2 directories.
foreach line (`awk '{print $1}' intflist`)
  cd $line
  echo $line
  ln -s ../mask_def.grd
  snaphu_interp.csh 0.01 0 0/8618/0/1839
  cd ..
end

and then ran unwrap_intf.csh in the /merge/ directory.

I've also included a screenshot of the ncview of my corr_stack:
![corr_stack](https://user-images.githubusercontent.com/61254441/76994976-8a1bb700-6925-11ea-9bd3-0917ed583b72.PNG)

Please let me know what other information can help.

Thanks,
Mieszko

**Comment by Xiaohua-Eric-Xu** [comment]
Hi Mieszko,
The numbers you put in for snaphu_interp.csh is the number of pixels I assume. However, what it asks for are range and azimuth. I think what you need is 0/68944/0/3678
Xiaohua(Eric)

**Comment by Mieszko-Fydrych** [comment]
Hi again,

I overlooked that mistake and just fixed it. I am rerunning the process and will update you on whether or not it worked.

Thanks a lot,
Mieszko

**Comment by Mieszko-Fydrych** [comment]
♥Hello Xiaohua,

Although I'm not getting the same error as before, I'm still having issues with the last step. I get a warning : 

> read phase and correlation grids ...
> 0.000000 0.000381 0.000000 -18.018379
> fill the G matrix ...
> run least-squares problem over 8618 by 1839 pixel (0) ...
> matrix is rank-deficient: 0
> 
> WARNING: rank is zero. Check scene.tab and intf.tab for possible duplicates.
> 
> 0.00 0.00 0.00 0.00 
> 0.00 0.00 0.00 0.00 
> 0.00 0.00 0.00 0.00 
> 0.00 0.00 0.00 0.00 
> 0.00 0.00 0.00 0.00 
> 0.00 0.00 0.00 0.00 
> 0.00 0.00 0.00 0.00 
> 0.00 0.00 0.00 0.00 
> 0.00 0.00 0.00 -0.00 
> 0.00 0.00 0.00 0.00

I checked for duplicates but did not find any. My scene and intf files are attached for reference.

What could be causing this issue is not a duplicate? I checked my unwrap.grd and it seems fine. 

Also, the last step of the tutorial wants you to make symbolic links from the ../intf_all/ folder. However, I only have that folder with my F1,F2 and F3 directories. Am I missing a folder or should I link it to the one in the F1 folder?

Thanks,
Mieszko

**Comment by Xiaohua-Eric-Xu** [comment]
Hi Mieszko,
Could you send us your scene.tab and intf.tab?
Xiaohua(Eric)

**Comment by Mieszko-Fydrych** [comment]
Hi Xiaohua,
I've attached them there.
[intf.txt](https://github.com/gmtsar/gmtsar/files/4356577/intf.txt)
[scene.txt](https://github.com/gmtsar/gmtsar/files/4356578/scene.txt)

Thanks,
Mieszko

**Comment by Xiaohua-Eric-Xu** [comment]
This is strange. Could you also attach the command you used to run sbas. i.e. the parameters

**Comment by Mieszko-Fydrych** [comment]
Hi Xiaohua,
I run sbas using

> sbas intf.tab scene.tab 10 5 8618 1839 -range  925360 -incidence 40 -wavelength 0.0554658 -smooth -rms -dem

I calculated the range using the formula you posted on the fourms, shown here:
http://gmt.soest.hawaii.edu/boards/6/topics/7951?r=8292#message-8292

Let me know what else I can do.

Thanks,
Mieszko

**Comment by Xiaohua-Eric-Xu** [comment]
Hi Mieszko,
Could you put a number after -smooth? e.g. 1.0
Cheers,
Xiaohua(Eric)

**Comment by Mieszko-Fydrych** [comment]
Hi again,

That was the solution. But I'm still wondering about the last step of the tutorial where it wants you to make symbolic links from the ../intf_all/ folder. However, I only have that folder with my F1,F2 and F3 directories. Am I missing a folder or should I link it to the one in the F1 folder?

Thank you for all of your help,
Mieszko

**Comment by Xiaohua-Eric-Xu** [comment]
Hi Mieszko,
I didn't quite get what you meant. What's the linking?
Xiaohua(Eric)

**Comment by Mieszko-Fydrych** [comment]
Hi Xiaohua,

In step 11, part c, subsection ii, it states to run the following in the terminal from the sbas folder:

> ln -s ../topo/trans.dat .
> ln -s ../intf_all/2017134_2017158/gauss_*

However, ../intf_all does not exist realtive to /sbas/. Should I change this path or am I missing a folder?

Thanks,
Mieszko

**Comment by Xiaohua-Eric-Xu** [comment]
Did you merge subswaths? If you did link trans.dat from your merge folder and find gauss_* in F?/intf_all/date1_date2/. These files are used for geocoding. i.e. proj_ra2ll.csh

**Comment by Mieszko-Fydrych** [comment]
Yes I did,
And thank you, I'll link those now.
Mieszko

**Comment by Mieszko-Fydrych** [comment]
Sorry to keep bugging you, but when I run:

> gmt grd2cpt vel_ll.grd -T= -Z -Cjet > vel_ll.cpt
I get the following error:

> grd2cpt: Error: Z-slice with dz = 0

When I check my vel_11.grd file, it shows a min and max for variable z as 0. The output for my proj_ra2ll is : 

> Sampling in geocoordinates with 50 meter pixels ...
> blockmedian: Provides 3, expects 3-column binary data
> blockmedian: Processing input table data
> blockmedian: W: 4.20833333333 E: 8.21527777778 S: 50.6416666667 N: 51.4041666667 n_columns: 5770 n_rows: 1830
> blockmedian: Input 3 columns via binary records using format fff
> blockmedian: Output 3 columns via binary records using format fff
> blockmedian: N read: 11718013 N used: 11718013 outside_area: 0 N cells filled: 4778539

What did I do wrong here?

Thanks,
Mieszko

**Comment by Xiaohua-Eric-Xu** [comment]
1) could you check your unwrap.grd-s and corr.grd-s to see whether thy look OK to you?
2) could you check your vel.grd in the sbas folder to see whether it came out as all zeros?
3) when you unwrap, what threshold did you use? I expect it to be something like 0.05.

**Comment by Mieszko-Fydrych** [comment]
My vel.grd file came out as all zeroes. And I used the threshold of 0.01 for my unwrapping.

**Comment by Xiaohua-Eric-Xu** [comment]
for your scene.tab, could you put them to chronological order?

**Comment by Mieszko-Fydrych** [comment]
My vel.grd still turns out as all 0's.

**Comment by Xiaohua-Eric-Xu** [comment]
Oh, I see. Could you change all the filenames to their ref_id/rep_id. i.e. for something like S1_20150102_ALL, change it to 2015001. I think the program reads in as numbers, not string.

**Comment by Mieszko-Fydrych** [comment]
Assuming the ref_id follows the order in the baseline_table.dat, I changed my scene.tab to the following:

> 2018001 1680
> 2018002 1728
> 2018003 1734
> 2018004 1770
> 2018005 1776

But still get an all 0's vel.grd. 

What else can I check?

Thanks,
Mieszko

**Comment by Xiaohua-Eric-Xu** [comment]
Hi Mieszko,
Please refer them to their real ref_id. Do this for both intf.tab and scene.tab.
I attached an example.
[scene.txt](https://github.com/gmtsar/gmtsar/files/4357091/scene.txt)
[intf.txt](https://github.com/gmtsar/gmtsar/files/4357092/intf.txt)

Xiaohua(Eric)

**Comment by Mieszko-Fydrych** [comment]
Hi Xiaohua,

I changed their id values to the ref_id values but still get a vel.grd of all 0's. I've attached my files. 
[scene.txt](https://github.com/gmtsar/gmtsar/files/4357298/scene.txt)
[intf.txt](https://github.com/gmtsar/gmtsar/files/4357299/intf.txt)

Could I still have used the wrong ref_ids?

Thanks,
Mieszko

**Comment by Xiaohua-Eric-Xu** [comment]
Does all the unwrap.grd look OK to you?
What about the disp_*.grd

**Comment by Mieszko-Fydrych** [comment]
My unwrap looks fine to me, there is an offset but that should be fine, as its noticed in the tutorial as well. A screen is provided below. And my disp_0000000.grd shows all 0's. 

![unw](https://user-images.githubusercontent.com/61254441/77125619-bc5c1000-6a1c-11ea-96fa-a97f514c3c1f.PNG)

Thanks,
Mieszko

**Comment by Xiaohua-Eric-Xu** [comment]
OK. This is really strange. What's the terminal output from your sbas ran.

**Comment by Mieszko-Fydrych** [comment]
Hi Xiaohua,

I'm not sure why, but when I ran it again to give you the output, I was able to generate more disp_*.grd files, and projected the radar coordinates successfully.

Thank you for all of your help. I really appreciate it.

Thanks again,
Mieszko

**Comment by Xiaohua-Eric-Xu** [comment]
OK. Maybe it was just bad luck. :)

**Comment by shivamthescienceguy** [comment]
> Oh, I see. Could you change all the filenames to their ref_id/rep_id. i.e. for something like S1_20150102_ALL, change it to 2015001. I think the program reads in as numbers, not string.

Hie Eric,
Could you please tell me how do I change names automatically from S1_20150102_ALL to 2015001 or vice versa, in scene.tab and in intflist etc.

**Comment by Xiaohua-Eric-Xu** [comment]
An example is `grep clock S1_20150102_ALL.PRM | awk '{print $3}' | awk -F. '{print $1}'`, write a `foreach` loop to create a column of new names, then use `paste` and `awk` to create a new scene.tab or intflist.

**Comment by ATDehkordi** [comment]
Hi everyone,

I am currently running GMTSAR on subswath 3 of the Sentinel-1 time series. Everything was progressing smoothly, and during the unwrapping process, I used a region cut. My command in unwrap_intf.csh was: snaphu_interp.csh 0.2 1 0/25000/4500/12000. All of the interferograms were also successfully unwrapped.

Now, I want to run SBAS with this command: sbas intf.tab scene.tab 143 37 3210 6094 -wavelength 0.0554658 -incidence 43.5 -range 930958 -rms -dem.

However, I encountered an error: "Dimension don't match! ../intf_all/2023000_2023012/corr.grd."

I suspect this is because the initial corr.grd file did not have the region cut applied. What can I do?
